### PR TITLE
ajax data source

### LINKED ISSUE #1682: Ajax/JSON data source
Expose a data source that can be configured with  a URL to a REST endpoint along with a mapping or metadata for populating data columns, that can be used to pull data directly to the client. 

Requirements:
* must integrate with blaze server endpoints #1635
* must be usable to simplify spectrogram example further

### PR DESCRIPTION
**Pull request by hhuuggoo**
issues: closes #1682 

also includes bugfix for bokeh server garbage collection

@bryevdv please review

### CODE REVIEW

File: bokehjs/src/coffee/common/base.coffee [hunk 1]
```diff
@@ -252,6 +253,8 @@ define [
     ColumnDataSource:         'source/column_data_source'
     ServerDataSource:         'source/server_data_source'
     BlazeDataSource:          'source/blaze_data_source'
+    AjaxDataSource:           'source/ajax_data_source'
```
Comment (bryevdv) [inline bokehjs/src/coffee/common/base.coffee]: let's keep these in alpha order

**Comment by bryevdv** [comment]
What doe the format for the REST response have to be? Just a JSON object with arrays for associated values?

**Comment by bryevdv** [comment]
@hhuuggoo mark as "ready" when ready for review. Looks good so far, just a minor comment and there should be some docs or a few words about what is expected from the REST endpoint somewhere.

**Comment by bryevdv** [comment]
This, combined with a "JS Action" is going to be brilliant.

File: bokehjs/src/coffee/source/ajax_data_source.coffee [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    setup : (plot_view, glyph) =>
+      @pv = plot_view
+      @update()
+      if @get('polling_interval')
+        @interval = setInterval(@update, @get('polling_interval'))
+
+    update : () =>
+      $.ajax(
+        dataType: 'json'
```
Comment (matdrapeau) [inline bokehjs/src/coffee/source/ajax_data_source.coffee]: I would suggest adding those properties:
ifModified: true,
async: false

**Comment by matdrapeau** [comment]
Not sure how that could be done, but being able to provide a function that could parse and format the result of the ajax call before pushing data would be nice. Cause the endpoint might not be giving the response in such format ```{x: value, y: value}```

**Comment by bryevdv** [comment]
@matdrapeau that is planned, I think it will be in 0.8.1 though.

**Comment by matdrapeau** [comment]
@hhuuggoo very nice btw!
I will look into doing something similar to your patch for websockets

File: bokehjs/src/coffee/source/ajax_data_source.coffee (hunk 2 above)
Comment (bryevdv) [inline bokehjs/src/coffee/source/ajax_data_source.coffee]: Let's get a little experience with this first then we can extend it with more knobs and dials later.

**Comment by bryevdv** [comment]
LGTM, merging.

**Comment by hhuuggoo** [comment]
@matdrapeau I am in favor of exposing isModified as a parameter on the data source -however I think async is a bad idea - won't that block the browser until the call returns?

**Comment by matdrapeau** [comment]
@hhuuggoo async=false is a bad idea you are right but I am concerned about the fact that setInterval might build up multiple ajax calls when there is network latencies or the actual query is dumping large dataset.
Either setTimeout or async=false might mitigate this problem but it also means the provided time period will be greater than what is expected.

**Comment by matdrapeau** [comment]
@hhuuggoo realtime updates: let say you want to do a line chart and append data to the actual line, this is not supported but might be a great improvement.

**Pull request merged**